### DISCUSSION TITLE
A small change in the Slovak language

### DISCUSSION THREAD

**Discussion by SoftVienka** [emsesp/EMS-ESP32#1636] [Ideas]
Dobrý deň
Píšem v slovenčine, lebo sa to týka hlavne užívateľov zo slovenska.
Mohol by niekto skompilovať a potom skontrolovať zmenu v slovenčine  v súbore index.ts.
Ide o zmenu skloňovania a neviem, či som to správne pochopil. Na obrázku je terajší stav.
Zmena je na riadku 98 až 103 v súbore index.ts.
Zostavil by som si to sám, ale od verzie 3.6... mi platformio v určitom bode hlási chybu 
(3.5.x prebehlo zostavenie bez problémov).

  NUM_DEVICES: '{num} Zariaden{{í|ie|ia|ia|í}}',
  NUM_TEMP_SENSORS: '{num} Teplotn{{ých|ý|é|é|ých}} snímač{{ov||e|e|ov}}',
  NUM_ANALOG_SENSORS: '{num} Analogov{ých|ý|é|é|ých}} snímač{{ov||e|e|ov}}',
  NUM_DAYS: '{num} d{{ní|eň|ní|ní|ní}}',
  NUM_SECONDS: '{num} sek{{únd|unda|undy|undy|únd}}',
  NUM_HOURS: '{num} hod{{ín|ina|iny|iny|ín}}',
  NUM_MINUTES: '{num} minú{{t|ta|ty|ty|t}}', 

![slovenčina](https://github.com/emsesp/EMS-ESP32/assets/20871723/188b4882-a3dc-4770-8d61-d48835b3d61a)

google translate:
Good day
I write in Slovak, because it mainly concerns users from Slovakia.
Could someone compile and then check the change in Slovak in the index.ts file.
It's a change of declension and I don't know if I got it right. The picture shows the current state.
The change is on line 98 to 103 in the index.ts file.
I would build it myself, but since version 3.6... platformio gives me an error at some point
(3.5.x built without problems).

**Comment by proddy** [comment]
> I would build it myself, but since version 3.6... platformio gives me an error at some point

which version of windows are you using? there's a trick to change regional settings to make PlatformIO work again

**Reply by MichaelDvP** [reply]
Yes, sounds like the windows utf8 issue, the fix is described here: https://github.com/emsesp/EMS-ESP32/pull/1514#issuecomment-1864303383

**Reply by SoftVienka** [reply]
VSC-PlatformIO has no problem (I think) with compiling Slovak. The problem is something else.

![error](https://github.com/emsesp/EMS-ESP32/assets/20871723/b4f565f9-919e-41f9-80db-f064bb2885c5)

W11:
Edition:	Windows 11 Home
Version:	23H2
Installation date:	‎21.‎9.‎2022
Operating system assembly:	22631.3155
Interface:	Windows Feature Experience Pack 1000.22684.1000.0

VSC:
Version: 1.86.2 (user setup)
Confirmation of changes: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:40:56.878Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22631

**Reply by MichaelDvP** [reply]
Yes, that is exactly how the windows issue with utf8 show up. Windows seems to switch codepage on some characters and after that the next path is not found.

**Reply by SoftVienka** [reply]
Sorry I didn't believe it. When I turned on "Beta: Unicode UTF-8..." then the project build was OK.
Now I can try the "Slovak language" modifications.
Well thank you.
